### PR TITLE
docs: tutorial for model registry

### PR DESCRIPTION
**Pull request by sidneyw**
## Description
Adds a concept level guide to the documentation to help users understand the new entities introduced by the Model Registry feature.

## Test Plan
Render and proofread the docs.

### CODE REVIEW

File: docs/tutorials/model-registry.txt [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+  # removes a top level key from the metadata
+  model.remove_metadata(["key"])
+  # results in {"metrics": { "test_set_acc": 0.97 } }
+
+
+Managing Model Versions
+-----------------------
+
+Once the model exists in determined, you can register checkpoints as a versions
```
Comment (neilconway) [inline docs/tutorials/model-registry.txt:133]: Determined

Comment (neilconway) [reply docs/tutorials/model-registry.txt:133]: "as a versions" -> grammar

File: docs/tutorials/model-registry.txt [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  # Metadata is merged with existing metadata.
+  model.add_metadata({"metrics": {"test_set_loss": 0.091}})
+  # results in {"key": "value", "metrics": { "test_set_loss": 0.091 } }
+
+  # Only top-level keys are merged. The following line will override all values
+  # underneath the "metrics" key.
+  model.add_metadata({"metrics": {"test_set_acc": 0.97}})
+  # results in {"key": "value", "metrics": { "test_set_acc": 0.97 } }
+
+  # removes a top level key from the metadata
```
Comment (neilconway) [inline docs/tutorials/model-registry.txt:125]: For consistency, capitalize the first letter of the comment and punctuate with a period.

File: docs/tutorials/model-registry.txt [hunk 3]
```diff
@@ -0,0 +1,245 @@
+.. _organizing-models:
+
+Organizing Models in the Model Registry
+=======================================
+
+In this tutorial, we focus on organization of checkpoints via the Determined
```
Comment (neilconway) [inline docs/tutorials/model-registry.txt:6]: Can we include some discussion here about _why_ you'd want to organize models in a model registry, some of the use-cases this feature is designed to support, etc.? (Or if that information appears in a different doc that is fine.)

File: docs/tutorials/model-registry.txt [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+representation of a model for illustration.
+
+.. code:: json
+
+  {
+    "name": "mnist_cnn",
+    "description": "a character recognition model",
+    "metadata": {
+      "dataset_url": "http://yann.lecun.com/exdb/mnist/",
+      "git_url": "http://github.com/user/repo",
```
Comment (neilconway) [inline docs/tutorials/model-registry.txt:28]: Trailing comma is not legal JSON.

File: docs/tutorials/model-registry.txt [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    }
+  }
+
+Models are created via the Determined Python API, REST API, or CLI. This guide
+will cover the Python and CLI methods. For information on the REST API see
+the Swagger API documentation.
```
Comment (neilconway) [inline docs/tutorials/model-registry.txt:34]: Can we link to the Swagger API docs?

File: docs/tutorials/model-registry.txt [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+In this tutorial, we focus on organization of checkpoints via the Determined
+Model Registry. Before reading this guide, we recommend that users first read
+the :ref:`use-trained-models` how to guide which describes how to interact
+with checkpoints.
+
+
+Managing Models
+---------------
+
+A model is a named set of checkpoints. Each model in Determined must have a
```
Comment (neilconway) [inline docs/tutorials/model-registry.txt:15]: "A model is a named set of checkpoints." -> I'd quibble with this slightly: if a model *is* a named set of checkpoints, it logically shouldn't include any additional metadata.

Should we also clarify the use of the term "model" in this doc and how that differs from, say, the "model definition" we talk about elsewhere in the docs/system?

File: docs/tutorials/model-registry.txt [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+          "code/const.yaml": "391",
+          "code/model_def.py": "4067",
+          "determined-keras-model.h5": "25224"
+        },
+        "metadata": {},
+        "framework": "tensorflow-1.14.0",
+        "format": "h5",
+        "metrics": { "...": "..." }
+      },
+      "version": 1,
```
Comment (neilconway) [inline docs/tutorials/model-registry.txt:172]: Trailing comma

File: docs/tutorials/model-registry.txt [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+
+.. code:: python
+
+  from determined.experimental import Determined
+
+  model = Determined().get_model("model_name")
+
+  latest_version = model.get_version()
+
+  specific_versions = model.get_version(3)
```
Comment (neilconway) [inline docs/tutorials/model-registry.txt:220]: `versions` or `version`?

File: docs/tutorials/model-registry.txt [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+
+  model = Determined().get_model("model_name")
+
+  model_versions = model.get_versions()
+
+The CLI equivalent is as follows.
+
+.. code:: bash
+
+  det model list_versions <model_name>
```
Comment (neilconway) [inline docs/tutorials/model-registry.txt:238]: This isn't a documentation issue, but elsewhere we use hyphens to separate words in CLI commands, not underscores. (e.g., `det experiment list-trials`).

File: docs/tutorials/model-registry.txt (hunk 3 above)
Comment (sidneyw) [reply docs/tutorials/model-registry.txt:6]: Will do! This is going to be the only doc apart from the autogenerated reference material so I will add an introduction covering use cases.

File: docs/tutorials/model-registry.txt (hunk 6 above)
Comment (sidneyw) [reply docs/tutorials/model-registry.txt:15]: We added the metadata at the model level to capture anything that is relevant for all versions of the model as shown in the example. Is there different language you would prefer to describe the relationship between a model and its versions?

File: docs/tutorials/model-registry.txt (hunk 9 above)
Comment (sidneyw) [reply docs/tutorials/model-registry.txt:238]: I'm changing the documentation now and will open a PR to change the underscore to a hyphen.

**Comment by sidneyw** [comment]
Documentation will be accurate once https://github.com/determined-ai/determined/pull/961 lands

**Pull request merged**